frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx

```diff
@@ -1481,13 +1493,45 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                 }
             }
         }
+
+        const checkCustomEventConversionGoalHasSessionIds = async (
+            { conversionGoal }: { conversionGoal: WebAnalyticsConversionGoal | null },
+            breakpoint: () => void
+        ): Promise<void> => {
+            if (!values.featureFlags[FEATURE_FLAGS.WEB_ANALYTICS_WARN_CUSTOM_EVENT_NO_SESSION]) {
+                return
+            }
+
+            if (!conversionGoal || !('customEventName' in conversionGoal) || !conversionGoal.customEventName) {
+                actions.setConversionGoalWarning(null)
+                return
+            }
+            const { customEventName } = conversionGoal
+            // check if we have any conversion events from the last week without sessions ids
+
+            const response = await hogqlQuery(
+                `select count() from events where timestamp >= now() - toIntervalHour(24) AND ($sessionId IS NULL OR $sessionId = '') AND event = {event}`,
```

Comment:
When is it preferable to run a query inline like this versus outsourcing it to the `<Query/>` component?